extensions/wikia/CommunityHeader/WikiButtons.class.php

```diff
@@ -0,0 +1,134 @@
+<?php
+namespace CommunityHeader;
+
+use \SpecialPage;
+
+class WikiButtons implements \Iterator {
+	private $buttons;
+
+	public function __construct() {
+		$wgUser = \F::app()->wg->User;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)